SConstruct

```diff
@@ -860,7 +860,12 @@ else:  # GCC, Clang
     common_warnings = []
 
     if methods.using_gcc(env):
-        common_warnings += ["-Wshadow", "-Wno-misleading-indentation"]
+        common_warnings += [
+            "-Wshadow",
+            "-Wno-misleading-indentation",
+            # For optimized Object::cast_to / object.inherits_from()
+            "-Wvirtual-inheritance",
```

Comment:
Does that mean we can't use any thirdparty header using this? I'm using FastNoise2 in my module, which uses virtual inheritance in a header. It triggers this warning and fails compilation with Werror.

Reply:
It just means the headers will need to be wrapped with pragmas disabling the warning. A few places in the repo are already doing this, though the D3D12 module is probably the most obvious example